crates/smart-config/src/error.rs

```diff
@@ -128,9 +135,15 @@ impl fmt::Display for ParseError {
             format!(" [origin: {}]", self.origin)
         };
 
+        let validation = if let Some(validation) = &self.validation {
+            format!("validation '{validation}' failed: ")
+        } else {
+            String::new()
+        };
+
         write!(
             formatter,
-            "error parsing {field} in `{config}` at `{path}`{origin}: {err}",
+            "error parsing {field} in `{config}` at `{path}`{origin}: {validation}{err}",
```

Comment:
Nit: It can be confusing to call validation error a parsing error. Probably it would be better to handle separately, e.g. we either can't parse the fied, or we successfully parsed the field but the value broken the validation rule.